fastlane/lib/fastlane/actions/app_store_build_info.rb

```diff
@@ -0,0 +1,195 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      LATEST_BUILD_NUMBER = :LATEST_BUILD_NUMBER
+      LATEST_VERSION_NUMBER = :LATEST_VERSION_NUMBER
+    end
+
+    class AppStoreBuildInfoAction < Action
+      def self.run(params)
+        require 'spaceship'
+
+        build_nr = get_build_number(params)
+
+        # Convert build_nr to int (for legacy use) if no "." in string
+        if build_nr.kind_of?(String) && !build_nr.include?(".")
+          build_nr = build_nr.to_i
+        end
+
+        Actions.lane_context[SharedValues::LATEST_BUILD_NUMBER] = build_nr
+      end
+
+      def self.get_build_number(params)
+        UI.message("Login to iTunes Connect (#{params[:username]})")
+        Spaceship::Tunes.login(params[:username])
+        Spaceship::Tunes.select_team(params[:team_id], params[:team_name])
+        UI.message("Login successful")
+
+        platform = params[:platform]
+
+        app = Spaceship::Tunes::Application.find(params[:app_identifier])
+        if params[:live]
+          UI.message("Fetching the latest build number for live-version")
+          live_version = app.live_version
+          build_nr = live_version.current_build_number
+          Actions.lane_context[SharedValues::LATEST_VERSION_NUMBER] = live_version.version
+        else
+          version_number = params[:version]
+          unless version_number
+            # Automatically fetch the latest version in testflight
+            begin
+              train_numbers = app.all_build_train_numbers(platform: platform)
+              testflight_version = self.order_versions(train_numbers).last
+            rescue
+              testflight_version = params[:version]
+            end
+
+            if testflight_version
+              version_number = testflight_version
+            else
+              version_number = UI.input("You have to specify a new version number, as there are multiple to choose from")
+            end
+
+          end
+
+          Actions.lane_context[SharedValues::LATEST_VERSION_NUMBER] = version_number
+
+          UI.message("Fetching the latest build number for version #{version_number}")
+
+          begin
+            build_numbers = app.all_builds_for_train(train: version_number, platform: platform).map(&:build_version)
+            build_nr = self.order_versions(build_numbers).last
+            if build_nr.nil? && params[:initial_build_number]
+              UI.message("Could not find a build on iTC. Using supplied 'initial_build_number' option")
+              build_nr = params[:initial_build_number]
+            end
+          rescue
+            UI.user_error!("Could not find a build on iTC - and 'initial_build_number' option is not set") unless params[:initial_build_number]
+            build_nr = params[:initial_build_number]
+          end
+        end
+        UI.message("Latest upload for version #{version_number} is build: #{build_nr}")
+
+        build_nr
```

Comment:
Probably should return a dictionary of info here? Build number, version number, etc?

Reply:
Just returning the same value as before. Other values can be accessed via SharedValues.

Reply:
I'm thinking since this is called `app_store_build_info` it should return a dictionary of all the info and then `app_store_build_number` should pull the build number from that. Having `app_store_build_info` only returning the build number just doesn't seem right 😇

Reply:
done.